cli/src/cmd/run.rs

```diff
@@ -89,168 +88,127 @@ impl Cmd for RunArgs {
             })
             .collect::<BTreeMap<String, (Abi, Vec<u8>)>>();
 
-        known_contracts.insert("VM".to_string(), (HEVM_ABI.clone(), Vec::new()));
-        known_contracts.insert("VM_CONSOLE".to_string(), (HEVMCONSOLE_ABI.clone(), Vec::new()));
-        known_contracts.insert("CONSOLE".to_string(), (CONSOLE_ABI.clone(), Vec::new()));
-
         let CompactContractBytecode { abi, bytecode, .. } = contract;
-        let abi = abi.expect("No abi for contract");
-        let bytecode = bytecode.expect("No bytecode").object.into_bytes().unwrap();
+        let abi = abi.expect("no ABI for contract");
+        let bytecode = bytecode.expect("no bytecode for contract").object.into_bytes().unwrap();
         let needs_setup = abi.functions().any(|func| func.name == "setUp");
 
-        let mut cfg = crate::utils::sputnik_cfg(&evm_version);
-        cfg.create_contract_limit = None;
-        let vicinity = evm_opts.vicinity()?;
-        let backend = evm_opts.backend(&vicinity)?;
-
-        // need to match on the backend type
-        let result = match backend {
-            BackendKind::Simple(ref backend) => {
-                let runner = ContractRunner::new(
-                    &evm_opts,
-                    &cfg,
-                    backend,
-                    &abi,
-                    bytecode,
-                    Some(evm_opts.sender),
-                    None,
-                    &predeploy_libraries,
-                );
-                runner.run_test(&func, needs_setup, Some(&known_contracts))?
-            }
-            BackendKind::Shared(ref backend) => {
-                let runner = ContractRunner::new(
-                    &evm_opts,
-                    &cfg,
-                    backend,
-                    &abi,
-                    bytecode,
-                    Some(evm_opts.sender),
-                    None,
-                    &predeploy_libraries,
-                );
-                runner.run_test(&func, needs_setup, Some(&known_contracts))?
-            }
+        let mut builder = ExecutorBuilder::new()
+            .with_cheatcodes(evm_opts.ffi)
+            .with_config(evm_opts.evm_env())
+            .with_spec(crate::utils::evm_spec(&config.evm_version));
+        if let Some(ref url) = self.evm_opts.fork_url {
+            let fork = Fork { url: url.clone(), pin_block: self.evm_opts.fork_block_number };
+            builder = builder.with_fork(fork);
+        }
+        if verbosity >= 3 {
+            builder = builder.with_tracing();
+        }
+        if evm_opts.debug {
+            builder = builder.with_tracing().with_debugger();
+        }
+
+        let mut result = {
+            let mut runner =
+                Runner::new(builder.build(), evm_opts.initial_balance, evm_opts.sender);
+            let (address, mut result) =
+                runner.setup(&predeploy_libraries, bytecode, needs_setup)?;
```

Comment:
I moved `forge run` away from using `ContractRunner` since I think that should be kept more test specific. We could move this runner into the forge crate if it makes sense, and maybe even build `ContractRunner` on top of it instead